src/EnergyPlus/SetPointManager.cc

```diff
@@ -241,31 +283,43 @@ void ManageSetPoints(EnergyPlusData &state)
         UpdateSetPointManagers(state);
         // The Mixed Air Setpoint Managers (since they depend on other setpoints, they must be calculated
         // and updated next to last).
-        for (SetPtMgrNum = 1; SetPtMgrNum <= state.dataSetPointManager->NumMixedAirSetPtMgrs; ++SetPtMgrNum) {
-            state.dataSetPointManager->MixedAirSetPtMgr(SetPtMgrNum).calculate(state);
+        for (auto *spm : state.dataSetPointManager->spms) {
+            if (spm->type == SPMType::MixedAir)
+                spm->calculate(state);
         }
         UpdateMixedAirSetPoints(state);
         // The Outside Air Pretreat Setpoint Managers (since they depend on other setpoints, they must be calculated
         // and updated last).
-        for (SetPtMgrNum = 1; SetPtMgrNum <= state.dataSetPointManager->NumOAPretreatSetPtMgrs; ++SetPtMgrNum) {
-            state.dataSetPointManager->OAPretreatSetPtMgr(SetPtMgrNum).calculate(state);
+        for (auto *spm : state.dataSetPointManager->spms) {
+            if (spm->type == SPMType::OutsideAirPretreat)
+                spm->calculate(state);
         }
+
         UpdateOAPretreatSetPoints(state);
     }
-}
+} // ManageSetPoints()
 
+int GetSetPointManagerIndex(EnergyPlusData &state, std::string const &Name)
+{
+    auto found = state.dataSetPointManager->spmMap.find(Name);
+    return (found != state.dataSetPointManager->spmMap.end()) ? found->second : 0;
+} // GetSetPointManagerIndex()
```

Comment:
So this is the reason all SPMs must have a unique name? Not that having unique names is a bad thing, or that users should want to have unique names across all SPMs, just that a name/type would always be unique if a unique name is used within each class. I am only thinking about this because of UnitarySystem now managing more parent object classes and whether there is anything I can do about not forcing the requirement of unique names (and the new getIndex function added to UnitarySystem in #10452). I think I am landing on unique names being a good thing going forward. Why would you name a PTAC the same as a CoilSystem? or a scheduled SPM the same as a mixed air SPM? In fact, it may be a good thing to force unique object names for every object in E+.

Reply:
Well, there are at least three things going on here. First, previously name uniqueness was not forced on SPMs at all. Not within an SPM sub-class, not across sub-classes. And I am guessing this is because some created a situation in which SPM input data was read twice and couldn't think of a way to resolve this issue other than disabling the check.

Second, I don't think we actually need to enforce name uniqueness within the instances of a certain IDF object because epJSON does that already. It uses the object names as keys and you can't have duplicate keys.

Third, the "convention" within EnergyPlus right now is to enforce name uniqueness within a "parent" class for lack of a better term. So, there is name uniqueness across all curves, and all schedules, and all chillers, etc. This is what "Global Names" is used for and this is what I did here. We currently do not enforce name uniqueness across an entire IDF file. We can start to do that, I am not sure how much that will help. Name uniqueness within the same class hierarchy is helpful because ultimately you are going to want to handle members of that hierarchy in a uniform way (so you can delete 65% of the code!) and that means storing them in the same data structures. If you have multiple objects of the same name in the same data structure, things can get a bit messy/verbose because then you need a secondary key everywhere to tell one from the other. But you're not going to run into that problem for objects that are not in the same class hierarchy. Having a Schedule and a Fan with the same name is not a problem internally because those two objects will never be handled by the same piece of code.